standards/stef_and_luka-standard_proposal_draft.md

```diff
@@ -0,0 +1,274 @@
+
+interface ERC948Write /* is ERC948 */ {
+/**
+   * @dev This emits when subscription is created.
+   **/
+  event Subscription(address indexed _user,
+                     address indexed _provider,
+                     uint256 indexed _subscriptionId);
+
+  /**
+   * @dev This emits when subscription is cancelled. _from address belongs to
+   * user or provider.
+   **/
+  event SubscriptionCancellation(address indexed _from,
+                                 address indexed _provider,
+                                 uint256 indexed _subscriptionId);
+
+  /**
+   * @dev This emits when recurring payment is executed. _from address belongs
+   * to provider (or processor).
+   **/
+  event SubscriptionPayment(address indexed _from,
+                            address indexed _provider,
+                            uint256 indexed _subscriptionId);
+
+  /**
+   * @dev Cancel active subscription. This can be done by user or provider.
+   * @param _provider Provider's address.
+   * @param _subscriptionId Subscription's ID.
+   **/
+  function cancelSubscription(address _provider, uint256 _subscriptionId) external;
+
+  /**
+   * @notice Called by provider.
+   * @dev Execute overdue payment.
+   * @param _provider Provider's address.
+   * @param _subscriptionId Subscription ID.
+   **/
+  function executePayment(address _provider, uint256 _subscriptionId) external;
+}
+```
+
+## ERC948Metadata Interface
+
+The metadata extension is OPTIONAL for ERC948 smart contracts. This allows a smart contract to be quieried about the subscription details.
+
+
+```solidity
+interface ERC948Metadata /* is ERC948 */ {
+
+  /**
+   * @notice A distinct Uniform Resource Identifier (URI) for a given subscription.
+   * @dev Throws if `_provider` and `_subscriptionID` are not a valid
+   * combination. URIs are defined in RFC 3986. The URI may point to a JSON file
+   * that conforms to the "ERC948 Metadata JSON Schema" (NOTE: NOT YET DEFINED).
+   * @param _provider Provider's address.
+   * @param _subscriptionId Subscription's ID.
+   **/
+    function subscriptionURI(address _provider, uint256 _subscriptionId)
+      external view returns (string);
+}
+```
+
+# Support for Processors
+
+## Short Summary
+
+Delegate payment executions to a third party.
+
+## Abstract
+
+Payment execution can be delegated to a third party subscription processors ("operators"). Providers/merchants want reliable recurring payments and might be willing to pay for that service.
+
+The extended standard offers subscription providers a way to register a third party subscription processor to execute payments on their behalf.
+
+## Motivation
+
+Subscription provider might not want to be the one executing payments since is does not want to deal with interacting with the subscription contract, following events, handling overdue payments and might want to delegate this responsibility to a third party.
+
+## Interface
+```solidity
+
+interface ERC948Processor /* is ERC948 */ {
```

Comment:
I think from the user perspective does not really matter who is executing the payment hence this part doesn't seem to be necessary. The only reason (maybe) to have it is to solve the coordination problem by assigning the processor for the payment.

Reply:
This is not part of the main standard but an extension aimed towards subscription providers who might want to delegate some of the responsibility.